pkg/conn/baseconn.go

```diff
@@ -114,11 +78,11 @@ func (conn *BaseConn) ExecuteSQLWithIgnoreError(tctx *tcontext.Context, ignoreEr
 	if len(queries) == 0 {
 		return 0, nil
 	}
-	if conn == nil || conn.DB == nil {
+	if conn == nil || conn.DBConn == nil {
 		return 0, terror.ErrDBUnExpect.Generate("database connection not valid")
 	}
 
-	txn, err := conn.DB.Begin()
```

Comment:
These conn interfaces, including `QuerySQL`, `ExecuteSQLWithIgnoreError` etc will be used as `operateFn` in RetryStrategy, so what will happen if the first call of `operateFn` is faild (maybe because the database connection is broken), then another retry will definitely fail.
btw it seems that our test cases don't cover the real database connection broken scenario

Reply:
well this not a problem, because we only retry on some specific errors now, not including database connection broken such as BadConn. no need to address

Reply:
but will we add the internal database connection management in this pr?
such as will we re-establish for the BadConn?

Reply:
good idea, let me try

Reply:
how about add [error](https://github.com/pingcap/dm/pull/266/files#diff-17b57264f15f5001c686d1232127256fR41) check in `retryFn` , and reset conn when get these error

Reply:
agree